test/Mono.Linker.Tests.Cases/Reflection/RunClassConstructorUsedViaReflection.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Reflection;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using Mono.Linker.Tests.Cases.Expectations.Metadata;
+using System.Runtime.CompilerServices;
+
+namespace Mono.Linker.Tests.Cases.Reflection
+{
+	[SetupCSharpCompilerToUse ("csc")]
+	public class RunClassConstructorUsedViaReflection
+	{
+		public static void Main ()
+		{
+			TestRunClassConstructor ();
+			TestNonKeptStaticConstructor ();
+			TestNull ();
+			TestDataFlowType ();
+			TestIfElse (1);
+		}
+
+		[Kept]
+		static void TestRunClassConstructor ()
+		{
+			RuntimeHelpers.RunClassConstructor (typeof (OnlyUsedViaReflection).TypeHandle);
+		}
+
+		[Kept]
+		static void TestNonKeptStaticConstructor ()
+		{
+			var a = new NonKeptStaticConstructorClass();
+		}
+
+		[Kept]
+		static void TestNull ()
+		{
+			Type type = null;
+			RuntimeHelpers.RunClassConstructor (type.TypeHandle);
+		}
+
+		[Kept]
+		static Type FindType ()
+		{
+			return null;
+		}
+
+		[Kept]
+		static void TestDataFlowType ()
+		{
+			Type type = FindType ();
+			RuntimeHelpers.RunClassConstructor (type.TypeHandle);
```

Comment:
This should warn - we have no idea what the type is. BTW: This should warn even with your changes.

The catch here is that the test won't validate the warnings unless the method has some related attribute on it (this is to avoid having to add the `Unrecognized` attributes all over the place). So to get the necessary verification you need to either add one of the `Unrecognized` attributes or add `RecognizedReflectionAccessPattern()` - this basically means "I expect this to not issue any warnings".